src/terra-dev-site/IssueForm/Index.issueform.jsx

```diff
@@ -0,0 +1,223 @@
+  const [issueType, setIssueType] = useState(initialState.issueType);
+  const [mentions, setMentions] = useState(initialState.mentions);
+  const [selectedPackage, setPackage] = useState(initialState.selectedPackage);
+  const [solution, setSolution] = useState(initialState.solution);
+  const [steps, setSteps] = useState(initialState.steps);
+  const [title, setTitle] = useState(initialState.title);
+
+  /**
+   * Initialize and track character count between updates and form changes.
+   * The total amount of characters that can be submitted in a URL is limited,
+   * so the form has an upper limit of 5500 characters. IE will fail at around 2K.
+   */
+  const total = issueType === 'bug'
+    ? [context, description, environment, expected, mentions, selectedPackage, solution, steps, title]
+      .reduce((prev, current) => prev + current).length
+    : [context, description, mentions, selectedPackage, title]
+      .reduce((prev, current) => prev + current).length;
+
+  // Basic test to check if form data has been entered when attempting to navigate away from the page.
+  const formDidUpdate = [title, description, steps, expected, context, mentions].reduce((prev, current) => prev + current).length > 0;
+  useEffect(() => {
+    setCount(total);
+    if (formDidUpdate) {
+      window.onbeforeunload = () => true;
+    } else {
+      window.onbeforeunload = undefined;
+    }
+  });
+
+  // Update repo based on the currently selected package.
+  const packageRepo = getRepo(selectedPackage);
+
+  // Construct error text template, with link to appropriate repo issue selection based on selected package.
+  const errorText = errorTemplate(packageRepo);
+
+  // Construct issue body based on current input form data.
+  const issueBody = issueType === 'bug'
+    ? bugTemplate({
+      description,
+      steps,
+      context,
+      expected,
+      solution,
+      environment,
+      mentions,
+    })
+    : featureTemplate({ description, context, mentions });
+
+  // Construct markdown template with current data for previewing.
+  const previewBody = titleTemplate(title, packageRepo, selectedPackage) + issueBody;
+
+  // Set target for preview window popup, and toggle open status.
+  const popupTarget = () => document.getElementById('preview-button');
+  function togglePopup() {
+    return !isOpen
+      ? setIsOpen(true)
+      : setIsOpen(false);
+  }
+
+  // Wait to ensure form validation has completed, then submit form data to the appropriate repo on github.
+  const submitForm = async () => {
+    await sleep(500);
+    const encodeTitle = encodeURIComponent(title).replace(/%2B/gi, '+');
+    const encodeBody = encodeURIComponent(issueBody).replace(/%2B/gi, '+');
+
+    window.open(
+      `https://github.com/cerner/${packageRepo}/issues/new?title=[${selectedPackage}] ${encodeTitle}&body=${encodeBody}`,
+    );
+  };
+
+  return (
+    <Spacer padding="large+2">
+      <Base>
+        <Markdown src={disclaimerTemplate} />
```

Comment:
Want to use `terra-heading` to title the page? Nice job with this!
```suggestion
        <Heading level={1}>Issue Form</Heading>
          <Markdown src={disclaimerTemplate} />
```

Reply:
Added: [f2442f8](https://github.com/cerner/terra-ui/pull/151/commits/f2442f885ea2592df2f3955cbd303d81cf590bf9), and thank you!